packages/gitbook-docs/src/parseDoc.ts

```diff
@@ -0,0 +1,103 @@
+import { pipe } from 'fp-ts/lib/function';
+import * as E from 'fp-ts/lib/Either';
+import * as RA from 'fp-ts/lib/ReadonlyArray';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export type DocSource<T> = T & {
+  readonly source: {
+    readonly pathPrefix: string;
+    readonly dirPath: string;
+  };
+};
+
+export type DocPage<T> = T & {
+  readonly page: {
+    readonly path: string;
+    readonly menu: string;
+    readonly body: string;
+  };
+};
+
+export type ParseDocEnv = {
+  readonly readFile: typeof fs.readFileSync;
+  readonly readDir: typeof fs.readdirSync;
+  readonly readStat: typeof fs.statSync;
+};
```

Comment:
again, not required to change now, but just to let you know, you could perform IO operations async ([check this out](https://nodejs.org/api/fs.html#promises-api))